src/main/java/de/jtheb/fh/lagerverwaltung/entities/Shelf.java

```diff
@@ -0,0 +1,66 @@
+package de.jtheb.fh.lagerverwaltung.entities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Shelf {
+    public static final int COMPARTMENTCOUNT = 100;
+
+    private List<Compartment> compartments = new ArrayList<>();
+
+    public Shelf() {
+        this.compartments = new ArrayList<>();
+    }
+
+    public boolean itemFits(Item item) {
+        if (!isFull()) {
+            for (Compartment compartment : compartments) {
+                if (compartment.itemFitsWithArticleNr(item) || compartment.getArticleNr().equals(null)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public boolean isFull() {
+        for (Compartment compartment : this.compartments) {
+            if (!compartment.isFull()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public Compartment findFittingCompartment(Item item) {
+        for (Compartment compartment : this.compartments) {
+            if (compartment.itemFitsWithArticleNr(item)) {
+                return compartment;
+            }
+        }
+        for (Compartment compartment : compartments) {
+            if (compartment.getArticleNr().equals(null)) {
+                return compartment;
+            }
+        }
+        throw new UnsupportedOperationException("Error - compartements full"); //TODO Correct ERROR Function/Shelffull
```

Comment:
UnsupportedOperationException doesnt fit. Create a custom exception. Ask me if you don't know how.